File: artemis-entity-tracker/src/net/namekdev/entity_tracker/model/ComponentTypeInfo.java
package net.namekdev.entity_tracker.model;

import java.util.Vector;


public class ComponentTypeInfo {
	/** Only available on server side. */
	public Class<?> type;

	public String name;
	public int index;
	public final Vector<FieldInfo> fields = new Vector<FieldInfo>();


	public ComponentTypeInfo(String name) {
		this.name = name;
	}

	public ComponentTypeInfo(Class<?> type) {
		this.type = type;
		this.name = type.getSimpleName();
	}
}

File: artemis-entity-tracker/src/main/java/net/namekdev/entity_tracker/model/ComponentTypeInfo.kt
package net.namekdev.entity_tracker.model

import net.namekdev.entity_tracker.utils.serialization.ObjectModelNode


class ComponentTypeInfo {
    /** Only available on server side.  */
    var type: Class<*>? = null

    var name: String
    var index: Int = 0
    lateinit var model: ObjectModelNode


    constructor(name: String) {
        this.name = name
    }

    constructor(type: Class<*>) {
        this.type = type
        this.name = type.simpleName
    }
}
